src/util/interpolate.ts

```diff
@@ -37,6 +38,10 @@ function number(from: number, to: number, t: number): number {
     return from + t * (to - from);
 }
 
+function projection(from: Projection, to: Projection, interpolation: number): Projection {
+    return new Projection(from, to, interpolation);
```

Comment:
```suggestion
    return new Projection(from.from, to.to, interpolation);
```
That way Projection can always hold a string in from and to?

Reply:
Doesn't this throw away the information of
- from.to
- from.transition
- to.from
- to.transiton

Reply:
It does, do we need it?

Reply:
only for very recursion - going between two transitions.

Reply:
We can probably disregard that usecase

Reply:
Much like color, we should be able to parse "globe" to be `["globe", "globe", 1]`, like red is converted to rgba under the hood.

Reply:
Check out the parse method of color here:
https://github.com/maplibre/maplibre-style-spec/blob/5788145317e1b7cf444a324207822706eec9071a/src/util/color.ts#L61

Can we do something similar for projection?
I'm expecting in the future to get wkt projection definition here in theory, so it would be good to have a parser that can support the simple types for now, not need to support wkt right now, but I think we should take that into consideration  in terms of design.

Reply:
I'm not sure still why we want to convert "mercator" to ["mercator", "mercator", 1], and then convert it back to "mercator" and use the value.